### ISSUE TITLE
Splitting strings produces different values than the reference Ruby library

### ISSUE THREAD

**Issue by jsmartt** [jg-rp/liquid#134]
The following template code produces different results in the Python and Ruby implementations:

```
{% assign items = "" | split: "," %}
```

The Ruby reference implementation of Liquid returns an empty array, while the Python implementation returns an array with a single empty string value.

```python
# Python
from liquid import Template

template = Template('{% assign items = "" | split: "," %}{{items | size}}')
print(template.render())  # "1"
```

```ruby
# Ruby
require 'liquid'
template = Liquid::Template.parse('{% assign items = "" | split: "," %}{{items | size}}')
puts template.render() # 0
```

Splitting a string with a delimiter of the same value of the string also produces different results.

Summary of split behavior:

| Template | Ruby Result | Python Result |
| - | - | - |
| `"" \| split: ""` |`[]` | `[]` |
| `"" \| split: ","` |`[]` | `[""]` |
| `"," \| split: ","` |`[]` | `["", ""]` |
| `"abc" \| split: ""` |`["a", "b", "c"]` | `["a", "b", "c"]` |
| `"abc" \| split: ","` |`["abc"]` | `["abc"]` |
| `"abc" \| split: "abc"` |`[]` | `["", ""]` |

There may be other differing edge cases I didn't thing of.

**Comment by jsmartt** [comment]
Thanks for the quick turnaround on this! I hadn't noticed the `liquid.future` filters or Environment before, but that's a nice modular approach to be able to pull changes like this in. 👍

**Comment by jg-rp** [comment]
Hi @jsmartt, thanks for raising this issue.

The new `split` filter (as described in #135 ) is available in Python Liquid version 1.10.2, now released. And an entry has been added to the [known issues page](https://jg-rp.github.io/liquid/known_issues#the-split-filter).
